lib/graphql/schema/argument.rb

```diff
@@ -121,6 +121,14 @@ def type
       # Used by the runtime.
       # @api private
       def prepare_value(obj, value)
+        # NOTE: This fixes the issue on queries.
+        # QUESTION: Should an InputObject #prepare method be called and then
+        # later the obj#prepare (GraphQL::Schema::Object) be called upon the
+        # InputObject prepared value? Or should it be one OR the another?
+        if value.respond_to?(:prepare)
```

Comment:
I think this a fine way to do it. It seems like, if we ignored one configuration or the other, it could be pretty confusing. And, it doesn't seem any more complicated to do both, is that right? 


What do you think about making an explicit type check? I'm worried about calling someone's `String#prepare` monkey patch or something. We could do this: 

```suggestion
        if value.is_a?(GraphQL::Schema::InputObject)
```

Reply:
342b3db882aa516608f46ef221f6083f5e3bdd6a